### PR TITLE
Add app service app extension tests

### PR DESCRIPTION
**Pull request by JackLivio**
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Build with tests on. Run `make test`

### Summary
Add simple unit tests for app service app extension's subscribe functions.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by jacobkeeler** [comment]
@JackLivio I would suggest merging develop into this branch, since the style script has changes, and it looks like there are build failures because of where this PR was based.

File: src/components/application_manager/rpc_plugins/app_service_rpc_plugin/test/app_service_app_extension_test.cc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+using ::testing::ReturnRef;
+using ::testing::AtLeast;
+using ::testing::NiceMock;
+using test::components::application_manager_test::MockApplication;
+using ::app_service_rpc_plugin::AppServiceAppExtension;
+using ::app_service_rpc_plugin::AppServiceRpcPlugin;
+using ::application_manager::plugin_manager::MockRPCPlugin;
+
+const std::string media = "MEDIA";
+const std::string nav = "NAVIGATION";
```
Comment (jacobkeeler) [inline src/components/application_manager/rpc_plugins/app_service_rpc_plugin/test/app_service_app_extension_test.cc]: Maybe include tests for an unknown service type as well?

**Comment by LuxoftAKutsan** [comment]
@JackLivio @jacobkeeler  this fix will be included to some kind of 5.1.2 release or it will be available already in 5.2 ?

**Comment by JackLivio** [comment]
@LuxoftAKutsan Hi the fix is targeted for develop so it will make it into master on the next minor or major release, which is probably 5.2.

**Comment by JackLivio** [comment]
Closed in favor of #3097

**Pull request closed without merging**